src/bundle.c

```diff
@@ -1019,13 +1021,222 @@ static int install_bundles(struct list *bundles, struct list **subs, struct mani
 	return ret;
 }
 
+static int lex_sort(const void *a, const void *b)
+{
+	const char *name1 = (char *)a;
+	const char *name2 = (char *)b;
+	return strcmp(name1, name2);
+}
+
+/* Get a list of files in a directory sorted by filename
+ * with their fullpath, returns NULL on error (errno set by
+ * opendir).
+ */
+static struct list *get_dir_files_sorted(char *path)
+{
+        DIR *dir = NULL;
+        struct dirent *ent = NULL;
+        struct list *files = NULL;
+
+        dir = opendir(path);
+        if (!dir) {
+                /* caller should use errno for detecting issues */
+                return NULL;
+        }
+
+        while ((ent = readdir(dir))) {
+                char *name = NULL;
+                if (ent->d_name[0] == '.') {
+                        continue;
+                }
+                string_or_die(&name, "%s/%s", path, ent->d_name);
+                files = list_prepend_data(files, name);
+        }
+
+        closedir(dir);
+
+        return list_sort(files, lex_sort);
+}
+
+struct alias_lookup {
+	char *alias;
+	struct list *bundles;
+};
+
+/* Deep free on alias_lookup structs
+ */
+static void free_alias_lookup(void *lookup)
+{
+        struct alias_lookup *l = (struct alias_lookup *)lookup;
+        free(l->alias);
+        list_free_list_and_data(l->bundles, free);
+}
+
+/* Get bundles for a given alias but if the alias is
+ * not actually an alias, return a list with just the
+ * bundle name
+ */
+static struct list *get_alias_bundles(struct list *alias_definitions, char *alias)
+{
+        struct list *bundles = NULL;
+        struct list *iter = alias_definitions;
+
+        while (iter) {
+                struct alias_lookup *lookup = (struct alias_lookup *)iter->data;
+                iter = iter->next;
+                if (strcmp(lookup->alias, alias) == 0) {
+                        bundles = list_deep_clone_strs(lookup->bundles);
+                        break;
+                }
+        }
+        if (!bundles) {
+                bundles = list_prepend_data(bundles, strdup_or_die(alias));
+        }
+
+        return bundles;
+}
+
+/* Parse an alias definition file for a list of bundles
+ * it specifies
```

Comment:
Can you add an example of what this file looks like in the comment?

Reply:
Pretty trivial example added, let me know if you want more there.

Reply:
nope that's perfect. Thank you.